src/Hl7.FhirPath/FhirPath/Expressions/SymbolTableInit.cs

```diff
@@ -346,7 +346,10 @@ private static IEnumerable<ITypedElement> runRepeat(Closure ctx, IEnumerable<Inv
                     newContext.SetIndex(ElementNode.CreateList(index));
                     index++;
 
-                    newNodes.AddRange(lambda(newContext, InvokeeFactory.EmptyArgs));
+                    var candidates = lambda(newContext, InvokeeFactory.EmptyArgs);
```

Comment:
Could you not use the Linq method Except? So:
```c#
 newNodes.AddRange(candidates.Except(fullResult, EqualityOperators.TypedElementEqualityComparer));
```

Reply:
Yes!